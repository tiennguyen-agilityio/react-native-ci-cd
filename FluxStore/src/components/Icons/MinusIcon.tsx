import Svg, {Path} from 'react-native-svg';
import {TouchableOpacity} from 'react-native';
import {memo} from 'react';

import {IconProps} from '@/interfaces';
import {useThemeStore} from '@/stores';

export const MinusIcon = memo(
  ({width = 6, height = 3, disabled = false, color, style, onPress}: IconProps) => {
    const {
      theme: {icon},
    } = useThemeStore();

    return (
      <TouchableOpacity
        testID="minus-icon"
        activeOpacity={0.8}
        onPress={onPress}
        disabled={disabled}
        style={style}>
        <Svg width={width} height={height} viewBox="0 0 6 3" fill="none">
          <Path
            stroke={color || icon.default}
            strokeWidth={1.5}
            d="M0.100098 1.31445L5.89475 1.31445"
          />
        </Svg>
      </TouchableOpacity>
    );
  },
);
