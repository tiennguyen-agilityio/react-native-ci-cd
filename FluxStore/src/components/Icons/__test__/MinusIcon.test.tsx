import {render, fireEvent, screen} from '@testing-library/react-native';

import {MinusIcon} from '../MinusIcon';

describe('MinusIcon', () => {
  const onPress = jest.fn();

  it('renders correctly (matches snapshot)', () => {
    const {toJSON} = render(<MinusIcon />);
    expect(toJSON()).toMatchSnapshot();
  });

  it('renders correctly (matches snapshot) with isActive', () => {
    const {toJSON} = render(<MinusIcon isActive />);
    expect(toJSON()).toMatchSnapshot();
  });

  it('calls onPress when touched and not disabled', () => {
    render(<MinusIcon onPress={onPress} />);

    fireEvent.press(screen.getByTestId('minus-icon'));

    expect(onPress).toHaveBeenCalledTimes(1);
  });

  it('does NOT call onPress when disabled', () => {
    render(<MinusIcon onPress={onPress} disabled />);

    fireEvent.press(screen.getByTestId('minus-icon'));

    expect(onPress).not.toHaveBeenCalledTimes(0);
  });
});
