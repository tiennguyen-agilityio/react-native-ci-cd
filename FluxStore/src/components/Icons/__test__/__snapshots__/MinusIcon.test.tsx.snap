// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MinusIcon renders correctly (matches snapshot) 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": false,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "opacity": 1,
    }
  }
  testID="minus-icon"
>
  <RNSVGSvgView
    align="xMidYMid"
    bbHeight={3}
    bbWidth={6}
    fill="none"
    focusable={false}
    height={3}
    meetOrSlice={0}
    minX={0}
    minY={0}
    style={
      [
        {
          "backgroundColor": "transparent",
          "borderWidth": 0,
        },
        {
          "flex": 0,
          "height": 3,
          "width": 6,
        },
      ]
    }
    vbHeight={3}
    vbWidth={6}
    width={6}
  >
    <RNSVGGroup
      fill={null}
      propList={
        [
          "fill",
        ]
      }
    >
      <RNSVGPath
        d="M0.100098 1.31445L5.89475 1.31445"
        fill={
          {
            "payload": 4278190080,
            "type": 0,
          }
        }
        propList={
          [
            "stroke",
            "strokeWidth",
          ]
        }
        stroke={
          {
            "payload": 2147483648,
            "type": 0,
          }
        }
        strokeWidth={1.5}
      />
    </RNSVGGroup>
  </RNSVGSvgView>
</View>
`;

exports[`MinusIcon renders correctly (matches snapshot) with isActive 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": false,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "opacity": 1,
    }
  }
  testID="minus-icon"
>
  <RNSVGSvgView
    align="xMidYMid"
    bbHeight={3}
    bbWidth={6}
    fill="none"
    focusable={false}
    height={3}
    meetOrSlice={0}
    minX={0}
    minY={0}
    style={
      [
        {
          "backgroundColor": "transparent",
          "borderWidth": 0,
        },
        {
          "flex": 0,
          "height": 3,
          "width": 6,
        },
      ]
    }
    vbHeight={3}
    vbWidth={6}
    width={6}
  >
    <RNSVGGroup
      fill={null}
      propList={
        [
          "fill",
        ]
      }
    >
      <RNSVGPath
        d="M0.100098 1.31445L5.89475 1.31445"
        fill={
          {
            "payload": 4278190080,
            "type": 0,
          }
        }
        propList={
          [
            "stroke",
            "strokeWidth",
          ]
        }
        stroke={
          {
            "payload": 2147483648,
            "type": 0,
          }
        }
        strokeWidth={1.5}
      />
    </RNSVGGroup>
  </RNSVGSvgView>
</View>
`;
